src/wireviz/wv_html.py

```diff
@@ -1,54 +1,91 @@
+    if metadata:
+        for item, contents in metadata.items():
+            if isinstance(contents, (str, int, float)):
+                replacements[f'<!-- %{item}% -->'] = html_line_breaks(str(contents))
+            elif isinstance(contents, Dict):  # useful for authors, revisions
+                for index, (category, entry) in enumerate(contents.items()):
+                    if isinstance(entry, Dict):
+                        replacements[f'<!-- %{item}_{index+1}% -->'] = str(category)
+                        for entry_key, entry_value in entry.items():
+                            replacements[f'<!-- %{item}_{index+1}_{entry_key}% -->'] = html_line_breaks(str(entry_value))
+
+        replacements['"sheetsize_default"'] = '"{}"'.format(metadata.get('template',{}).get('sheetsize', ''))  # include quotes so no replacement happens within <style> definition
+
+    # perform replacements
+    # regex replacement adapted from:
+    # https://gist.github.com/bgusach/a967e0587d6e01e889fd1d776c5f3729
+    replacements_sorted = sorted(replacements, key=len, reverse=True)  # longer replacements first, just in case
+    replacements_escaped = map(re.escape, replacements_sorted)
+    pattern = re.compile("|".join(replacements_escaped))
+    html = pattern.sub(lambda match: replacements[match.group(0)], html)
+
     with open_file_write(f'{filename}.html') as file:
-        file.write('<!DOCTYPE html>\n')
-        file.write('<html lang="en"><head>\n')
-        file.write(' <meta charset="UTF-8">\n')
-        file.write(f' <meta name="generator" content="{APP_NAME} {__version__} - {APP_URL}">\n')
-        file.write(f' <title>{metadata["title"]}</title>\n')
-        file.write(f'</head><body style="font-family:{options.fontname};background-color:'
-                   f'{wv_colors.translate_color(options.bgcolor, "HEX")}">\n')
-
-        file.write(f'<h1>{metadata["title"]}</h1>\n')
-        description = metadata.get('description')
-        if description:
-            file.write(f'<p>{description}</p>\n')
-        file.write('<h2>Diagram</h2>\n')
-        with open_file_read(f'{filename}.svg') as svg:
-            file.write(re.sub(
-                '^<[?]xml [^?>]*[?]>[^<]*<!DOCTYPE [^>]*>',
-                '<!-- XML and DOCTYPE declarations from SVG file removed -->',
-                svg.read(1024), 1))
-            for svgdata in svg:
-                file.write(svgdata)
-
-        file.write('<h2>Bill of Materials</h2>\n')
-        listy = flatten2d(bom_list)
-        file.write('<table style="border:1px solid #000000; font-size: 14pt; border-spacing: 0px">\n')
-        file.write(' <tr>\n')
-        for item in listy[0]:
-            file.write(f'  <th style="text-align:left; border:1px solid #000000; padding: 8px">{item}</th>\n')
-        file.write(' </tr>\n')
-        for row in listy[1:]:
-            file.write(' <tr>\n')
-            for i, item in enumerate(row):
-                item_str = item.replace('\u00b2', '&sup2;')
```

Comment:
I cannot find this replacement in your new code. Is it obsolete and no longer needed, or handled in a different way?